### DISCUSSION TITLE
Replace custom certificates instead of creating new ones

### DISCUSSION THREAD

**Discussion by lug-gh** [NginxProxyManager/nginx-proxy-manager#2317] [Ideas]
If a self-uploaded custom certificate expires, you need to create a new one and then select the new certificate in the settings for each individual proxy host. Depending on the number of hosts, this is quite annoying, and also relatively unnecessary.
It would be better if you can replace the existing certificate with the new one instead of deleting the old one and uploading a new one.

My workaround is to exchange the certificate in the file system, but then in the web interface the old metadata (created, expires) remains, and doesn't seem to be updated. But I save myself hundreds of clicks :)

Additionally a housekeeping would not be wrong. Deleted certificates from the web interface remain in the file system.

**Comment by geozeke** [comment]
This is a great idea!

**Comment by SSpt1978** [comment]
That's a need for me

**Comment by begunfx** [comment]
Upvote here too!

**Comment by jmtremblay2** [comment]
any update on this?

**Reply by lug-gh** [reply]
> any update on this?

The further development of NPM version 2 was discontinued a long time ago.
See https://github.com/NginxProxyManager/nginx-proxy-manager/issues/1157

**Reply by CamelT0E** [reply]
This doesn't look any better with NPM v3. 
i think that this version will not appear and v2 will not get any new features.

**Comment by PatCas71** [comment]
Would love to see this feature too!
